networkit/cpp/graph/Graph.cpp

```diff
@@ -496,22 +498,36 @@ edgeweight Graph::computeMaxWeightedDegree(const bool inDegree) const {
     return result;
 }
 
-edgeweight Graph::computeWeightedDegree(const node &v,
-                                        const bool inDegree) const {
+edgeweight Graph::computeWeightedDegree(node u, bool inDegree, bool countSelfLoopsTwice) const {
     if (weighted) {
         edgeweight sum = 0.0;
-        auto sumWeights = [&](node, edgeweight w) { sum += w; };
+        auto sumWeights = [&](node v, edgeweight w) {
+            sum += (countSelfLoopsTwice && u == v) ? 2. * w : w;
+        };
         if (inDegree) {
-            forInNeighborsOf(v, sumWeights);
+            forInNeighborsOf(u, sumWeights);
         } else {
-            forNeighborsOf(v, sumWeights);
+            forNeighborsOf(u, sumWeights);
         }
         return sum;
     }
-    return defaultEdgeWeight * (inDegree ? degreeIn(v) : degreeOut(v));
+
+    count sum = inDegree ? degreeIn(u) : degreeOut(u);
+    auto countSelfLoops = [&](node v) { sum += (u == v); };
+
+    if (countSelfLoopsTwice) {
```

Comment:
It might be a good idea to add a (constant time) check if there are any self-loops (in the whole graph) here.

Reply:
That's a good idea, thank you. I've included it in the PR.